lib/spack/spack/directory_layout.py

```diff
@@ -430,6 +431,11 @@ def _extension_map(self, spec):
     def _write_extensions(self, spec, extensions):
         path = self.extension_file_path(spec)
 
+        if not extensions:
```

Comment:
(question) Is this required, or cosmetic?

Reply:
It's somewhere in between. The extensions.yaml files will show up as unknown files under `spack verify` and I think that will confuse admins who see it. It's one thing to see an `extensions.yaml` file when you know you have active extensions, it's another to see it when you have none.